src/sentry/debug_files/artifact_bundles.py

```diff
@@ -34,14 +50,374 @@
+        except ArtifactBundleFlatFileIndex.DoesNotExist:
+            return None
+
+    @classmethod
+    def _deserialize_json_index(
+        cls, json_index: Dict[str, Any]
+    ) -> Dict[str, Union[Bundles, FilesByUrl, FilesByDebugID]]:
+        deserialized_index: Dict[str, Union[Bundles, FilesByUrl, FilesByDebugID]] = {}
+
+        if "bundles" in json_index:
+            deserialized_index["bundles"] = cls._deserialize_bundles(json_index["bundles"])
+
+        if "files_by_url" in json_index:
+            deserialized_index["files_by_url"] = json_index["files_by_url"]
+
+        if "files_by_debug_id" in json_index:
+            deserialized_index["files_by_debug_id"] = json_index["files_by_debug_id"]
+
+        return deserialized_index
+
+    @classmethod
+    def _deserialize_bundles(cls, bundles: List[Dict[str, Union[int, str]]]) -> Bundles:
+        deserialized_bundles: Bundles = []
+
+        for bundle in bundles:
+            bundle_id = bundle["id"]
+            bundle_timestamp = bundle["timestamp"]
+
+            deserialized_bundles.append(
+                BundleMeta(
+                    id=int(bundle_id), timestamp=datetime.fromisoformat(str(bundle_timestamp))
+                )
+            )
+
+        return deserialized_bundles
+
+    def save(self, flat_file_index: Dict[str, Union[Bundles, FilesByUrl, FilesByDebugID]]):
+        serialized_index = self._serialize_json_index(flat_file_index)
+
+        ArtifactBundleFlatFileIndex.create_flat_file_index(
+            project_id=self.identifier.project_id,
+            release=self.identifier.release,
+            dist=self.identifier.dist,
+            file_contents=serialized_index,
+        )
+
+    @classmethod
+    def _serialize_json_index(
+        cls, flat_file_index: Dict[str, Union[Bundles, FilesByUrl, FilesByDebugID]]
+    ) -> str:
+        pre_serialized_index: Dict[str, Any] = {}
+
+        if "bundles" in flat_file_index:
+            pre_serialized_index["bundles"] = cls._serialize_bundles(
+                cast(Bundles, flat_file_index["bundles"])
+            )
+
+        if "files_by_url" in flat_file_index:
+            pre_serialized_index["files_by_url"] = flat_file_index["files_by_url"]
+
+        if "files_by_debug_id" in flat_file_index:
+            pre_serialized_index["files_by_debug_id"] = flat_file_index["files_by_debug_id"]
+
+        return json.dumps(pre_serialized_index)
+
+    @classmethod
+    def _serialize_bundles(cls, bundles: Bundles) -> List[Dict[str, Union[int, str]]]:
+        serialized_bundles: List[Dict[str, Union[int, str]]] = []
+
+        for bundle in bundles:
+            serialized_bundles.append(
+                {"id": bundle["id"], "timestamp": datetime.isoformat(bundle["timestamp"])}
+            )
+
+        return serialized_bundles
+
+
+T = TypeVar("T")
+
+
+class FlatFileIndex:
+    def __init__(self, store: FlatFileIndexStore):
+        self.store = store
```

Comment:
The `store` is only used for loading from and saving to the backing store.
This is the concern of the called IMO.